projects/chaos/inc/sequencer.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "chaos_events.h"
+#include "event_queue.h"
+#include "status.h"
+
+// Sequencer module for sequencing events.
+//
+// Expects event_queue to be initialized.
+//
+// A Sequencer is initialized with an array of SequencerEventPairs containing |raise|, |response|
+// pairs. |sequencer_init()| acts by raising the first |raise| event in the array and henceforth
+// |sequencer_advance()| is to be called with the previously raised event until the array is
+// consumed. If a |response| is event not |SEQUENCER_NO_OP| then once the previously raised event
+// is received it awaits |response|.
+
+#define SEQUENCER_EMPTY_DATA 0
```

Comment:
I'm not sure if this is strictly necessary since it should automatically zero fields that aren't explicitly set. Also, I think something like `NO_RESPONSE` is more accurate than `NO_OP` now?

Reply:
Done. It isn't necessary but I'd like to keep it explicit.